common/transport/amqp/src/receiver_link.ts

```diff
@@ -0,0 +1,208 @@
+import * as machina from 'machina';
+import * as amqp10 from 'amqp10';
+import * as dbg from 'debug';
+import { EventEmitter } from 'events';
+import { Message, errors, results } from 'azure-iot-common';
+import { AmqpMessage } from './amqp_message';
+import { AmqpLink } from './amqp_link_interface';
+
+const debug = dbg('ReceiverLink');
+
+/*Codes_SRS_NODE_AMQP_RECEIVER_LINK_16_002: [** The `ReceiverLink` class shall inherit from `EventEmitter`.]*/
+/*Codes_SRS_NODE_AMQP_RECEIVER_LINK_16_003: [** The `ReceiverLink` class shall implement the `AmqpLink` interface.]*/
+export class ReceiverLink  extends EventEmitter implements AmqpLink {
```

Comment:
>ReceiverLink [](start = 13, length = 12)

So whoever is using this ReceiverLink has no way of knowing that it got detached?